Sources/Sentry/SentryTracer.m

```diff
@@ -267,8 +272,12 @@ - (void)deadlineTimerFired
 
 - (void)cancelDeadlineTimer
 {
-    [self.deadlineTimer invalidate];
-    self.deadlineTimer = nil;
+    // The timer must be invalidated from the thread on which the timer was installed, see
+    // https://developer.apple.com/documentation/foundation/nstimer/1415405-invalidate#1770468
+    [_configuration.dispatchQueueWrapper dispatchOnMainQueue:^{
+        [self.deadlineTimer invalidate];
+        self.deadlineTimer = nil;
```

Comment:
Can this cause a retain cycle? I guess we sort of need it to hang around so we can actually cancel the timer, but will the tracer be released after this block exits?

Reply:
Yes, it definitely could. When the main thread is busy, the tracer is not deallocated as long as this block runs on the main thread. Good catch thanks 🥇. I fixed the code and added a test.